### PR TITLE
value len hex parse

### PR DESCRIPTION
**Pull request by Kritner**
#110

### CODE REVIEW

**Comment by bfussell** [comment]
I assume that a valueLen of 5 means the least significant 5 bits ? 
valueLen: 5, value: "FA" is 11010

May be good to point out.

**Comment by celic** [comment]
For most algorithms it is big endian. So in your example, the binary should be "11111". In accordance with issue #109 SHA3 is the only algorithm that is little endian where "FA" would correspond to "11010".

**Comment by atvassilev** [comment]
@celic this is a good point and implies that the valueLen property must also be defined for every algorithm. One approach could be to indicate the endianness problem (for some algorithms it is the LSB part for others it is the MSB part) in the protocol spec and warn the reader to check the algorithm spec for endianness definition. Then each algorithm spec, e.g. SHA3, would define which parts of the bit string to take. 

The bottom line would be to include a couple of examples illustrating the different endians and a warning in the text here for the user to check the corresponding algorithm spec for the endianness.

**Comment by celic** [comment]
Yes, this (should) become apparent when you look at the examples. If it doesn't then there should be additional examples. If you see hex such as "80" for length 1, it is clear the leftmost bit is the significant bit. On the other end, "01" for length 1 shows the opposite. It definitely should also be stated in the specification (ideally as you introduce the length or value the lengths refers to.) As far as I'm aware, SHA3 is the only algorithm that uses little endian hex.

**Comment by celic** [comment]
There is no reference to endianness in this update. The example given "FF" is ambiguous towards endianness as well. Can this be fixed before this is merged?

**Comment by Kritner** [comment]
Ah, I had intentionally left it ambiguous as it was in a general section of the specification, and I thought it was at least on our radar elsewhere for getting more specific with endianness.  

Can make changes (probably tomorrow - need to go drop off my car at the dealership before they close)

**Comment by celic** [comment]
Thanks. As it was brought up in this thread I figured it be worth it to mention in the document now.
